src/rtctools_heat_network/workflows/goals/minimize_tco_goal.py

```diff
@@ -1,69 +1,149 @@
-            *optimization_problem.heat_network_components.get("source", []),
-            *optimization_problem.heat_network_components.get("ates", []),
-            *optimization_problem.heat_network_components.get("buffer", []),
-        ]:
-            obj += (
-                optimization_problem.extra_variable(
-                    optimization_problem._asset_fixed_operational_cost_map[asset]
-                )
-                * self.number_of_years
-            )
+        This method sums up the various costs associated with the heat network assets.
 
-        for asset in [
-            *optimization_problem.heat_network_components.get("source", []),
-            *optimization_problem.heat_network_components.get("ates", []),
-            *optimization_problem.heat_network_components.get("buffer", []),
-            *optimization_problem.heat_network_components.get("demand", []),
-            *optimization_problem.heat_network_components.get("heat_exchanger", []),
-            *optimization_problem.heat_network_components.get("heat_pump", []),
-            *optimization_problem.heat_network_components.get("pipe", []),
-        ]:
-            obj += optimization_problem.extra_variable(
-                optimization_problem._asset_investment_cost_map[asset]
-            )
+        Args:
+            optimization_problem (TechnoEconomicMixin): The optimization problem instance.
+            ensemble_member: The current ensemble member being considered in the optimization.
+
+        Returns:
+            MX object: CasADi expression with the total cost objective function value.
+        """
 
-        for asset in [
-            *optimization_problem.heat_network_components.get("source", []),
-            *optimization_problem.heat_network_components.get("ates", []),
-            *optimization_problem.heat_network_components.get("buffer", []),
-            *optimization_problem.heat_network_components.get("demand", []),
-            *optimization_problem.heat_network_components.get("heat_exchanger", []),
-            *optimization_problem.heat_network_components.get("heat_pump", []),
-            *optimization_problem.heat_network_components.get("pipe", []),
-        ]:
-            obj += optimization_problem.extra_variable(
-                optimization_problem._asset_installation_cost_map[asset]
+        options = optimization_problem.heat_network_options()
+
+        cost_type_maps = {
+            "operational": optimization_problem._asset_variable_operational_cost_map,
+            "fixed_operational": optimization_problem._asset_fixed_operational_cost_map,
+            "investment": optimization_problem._asset_investment_cost_map,
+            "installation": optimization_problem._asset_installation_cost_map,
+            "annualized": optimization_problem._annualized_capex_var_map,
+        }
+
+        if options["discounted_annualized_cost"]:
+            cost_type_list = ["operational", "fixed_operational", "annualized"]
```

Comment:
Please correct me if I am wrong, but I think using the annualized only makes sense if we use a single year of OPEX costs, right? If so, can we do a check here to ensure that self.number_of_years==1?

Reply:
I think I see that number of years is ignored in _calculate_cost() when using annualized, correct?

Reply:
Hi @jimrojerTNO, 
Indeed, the underlyting assumption in this calculation for the annualized cost is that the operational and fixed_operational costs are already computed as annual values. 
As in the current implementation of the `MinimizeTCO ` goal the `number_of_years` is used to multiply the operational and fixed_operational costs, I conclude that the operational and fixed operational costs are already computed as annual values. For this reason, when `options["discounted_annualized_cost"] = True`, we don't need to verify `self.number_of_years==1` (as we don't use this parameter for the discounted calculation in this new implementation.

Reply:
In fact, for the discounted cost calculation, `self.number_of_years' is ignored. This is because we use the technical life of each asset instead.